### PR TITLE
dotnet 6.0.102

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Re-attempting v6.0.102. Originally tried in #92991 but hit some issues.

`do not merge` as need to check with upstream on workaround added.

Odd tag name due to: https://github.com/dotnet/source-build/discussions/2761

### CODE REVIEW

File: Formula/dotnet.rb [hunk 1]
```diff
@@ -41,22 +41,6 @@ class Dotnet < Formula
...
-  # PR ref: https://github.com/dotnet/installer/pull/12642
-  # TODO: Remove this in the next release
-  patch do
-    url "https://github.com/dotnet/installer/commit/7f02ccd30f55e7ac3436bd32af4b207869541ebf.patch?full_index=1"
-    sha256 "ff51630f9bfc4bbb502f57c6b1348d2e530a006234150606f9327fedcbb6591c"
-  end
-
   # Fix build failure on macOS due to missing ILAsm/ILDAsm
   # Fix build failure on macOS ARM due to `osx-x64` override
   patch :DATA
```
Comment (cho-m) [inline Formula/dotnet.rb:46]: **TODO**: Still need to raise issues and open PRs upstream.

File: Formula/dotnet.rb [hunk 2]
```diff
@@ -68,19 +52,16 @@ def install
...
-      # TODO: Remove this in the next release
-      inreplace Dir["src/runtime.*/eng/packaging.targets"].first,
-                "<EnablePackageValidation>true</EnablePackageValidation>",
-                "<EnablePackageValidation>false</EnablePackageValidation>"
+      # Workaround for error MSB4018 while building 'installer in tarball' due
+      # to trying to find aspnetcore-runtime-internal v6.0.0 rather than current.
+      # TODO: Remove when packaging is fixed
+      inreplace Dir["src/installer.*/src/redist/targets/GenerateLayout.targets"].first,
+                "$(MicrosoftAspNetCoreAppRuntimePackageVersion)",
+                "$(MicrosoftAspNetCoreAppRuntimewinx64PackageVersion)"
```
Comment (cho-m) [inline Formula/dotnet.rb:64]: Before merge, main thing to check upstream is why `MicrosoftAspNetCoreAppRuntimePackageVersion` is using **6.0.0** and failing.

`MicrosoftAspNetCoreAppRuntimewinx64PackageVersion` is set to **6.0.2** and it looks like build is supposed to use `MicrosoftAspNetCoreAppRuntimePackageVersion = MicrosoftAspNetCoreAppRuntimewinx64PackageVersion`

**Comment by cho-m** [comment]
I think this can be merged as is for now. I created upstream issue dotnet/source-build#2795 to track our patches/workarounds and find out how to fix upstream.

Only odd thing is new workaround where `MicrosoftAspNetCoreAppRuntimePackageVersion` is not properly set. For now, workaround works.

I believe the values are supposed to be the same based on https://github.com/dotnet/installer/blob/v6.0.102-source-build/eng/Versions.props#L86-L93
```
  <PropertyGroup>
    <!-- Runtime and Apphost pack versions are the same for all RIDs. We flow the x64    -->
    <!-- version above and create aliases without the winx64 here for clarity elsewhere. -->
    <MicrosoftNETCoreAppHostPackageVersion>$(MicrosoftNETCoreAppHostwinx64PackageVersion)</MicrosoftNETCoreAppHostPackageVersion>
    <MicrosoftNETCoreAppRuntimePackageVersion>$(MicrosoftNETCoreAppRuntimewinx64PackageVersion)</MicrosoftNETCoreAppRuntimePackageVersion>
    <MicrosoftAspNetCoreAppRuntimePackageVersion>$(MicrosoftAspNetCoreAppRuntimewinx64PackageVersion)</MicrosoftAspNetCoreAppRuntimePackageVersion>
    <MicrosoftWindowsDesktopAppRuntimePackageVersion>$(MicrosoftWindowsDesktopAppRuntimewinx64PackageVersion)</MicrosoftWindowsDesktopAppRuntimePackageVersion>
  </PropertyGroup>
```

**Comment by Bo98** [comment]
Fine for me. Did the just released 6.0.201 change anything?

**Comment by cho-m** [comment]
I didn't take a look at release notes yet.

I had problems building locally where build stalled. May try again or just run on CI.

**Comment by Bo98** [comment]
(For future reference of why 6.0.201 won't work: https://github.com/dotnet/source-build#support)

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**